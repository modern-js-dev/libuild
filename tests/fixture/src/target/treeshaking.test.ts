import { getLibuilderTest } from '@modern-js/libuild-test-toolkit';

describe('fixture:target', () => {
  it('2015', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      target: 'es2015',
      input: {
        es2015: './index.ts',
      },
    });
    await bundler.build();
    bundler.expectJSOutputMatchSnapshot();
  });
  it('2017', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      input: {
        es2017: './index.ts',
      },
      target: 'es2017',
    });
    await bundler.build();
    bundler.expectJSOutputMatchSnapshot();
  });
});
