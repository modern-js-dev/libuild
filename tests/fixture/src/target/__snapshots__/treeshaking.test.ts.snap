// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixture:target 2015 1`] = `
"var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// index.ts
function answer() {
  return __async(this, null, function* () {
    return 42;
  });
}
export {
  answer
};
"
`;

exports[`fixture:target 2017 1`] = `
"// index.ts
async function answer() {
  return 42;
}
export {
  answer
};
"
`;
